ocs_ci/ocs/cluster.py

```diff
@@ -912,6 +912,74 @@ def get_osd_utilization():
     return osd_filled
 
 
+def get_ceph_df_detail():
+    """
+    Get ceph osd df detail
+    :return: dict returned from cmd
+    """
+    ceph_cmd="ceph df detail"
+    ct_pod = pod.get_ceph_tools_pod()
+    output = ct_pod.exec_ceph_cmd(ceph_cmd=ceph_cmd)
+    return output
+
+
+def validate_replica_data(pool_name, replica):
+    """
+    Check if data is replica 2 or 3
+    :param int replica: size of the replica(2,3)
+    :param string pool_name: name of the pool to check replica
+    :return: true if replica is verified, false if not
+    """
+    from ocs_ci.ocs.exceptions import PoolNotFound
+    output = get_ceph_df_detail()
+    for pool in output.get('pools'):
+        if pool.get('name') == pool_name:
+            logger.info(f"{pool_name}")
+            stored = pool.get('stats').get('stored')
+            byte_used = pool.get('stats').get('bytes_used')
+            compress_bytes_used = pool.get('stats').get('compress_bytes_used')
+            compress_under_bytes = pool.get('stats').get('compress_under_bytes')
```

Comment:
This is a general complain, so feel free to ignore it, but I don't understand the approach of using `get()` method to fetch value in a dictionary everywhere. This will make sense only if we actually care to handle the case when it returns None. If we don't care (eg. because a test or function should fail because of the missing key), using `get` will make the traceback more opaque, noting "None doesn't have get method", instead of "dict such and such doesn't have "stats" key.

Reply:
Do you mean using 
pool['stats']['compress_bytes_used'] ? This will make more efficient to debug?

Reply:
Yes.

Reply:
That is because when this fails, exception will look like `KeyError: 'bytes_used'` on given dict, which will immediately tell us that 'bytes_used' key is not present in given dict. Compare that with exception `AttributeError: 'NoneType' object has no attribute 'get'` which doesn't tell us much at first sight.

And off curse, if you can recover from the error, it makes sense to use `.get()` methods, but then you also need to check whether the value is None, and log and process the error properly.

Here I agree that it doesn't make much sense to check whether the keys are there or not, because this code can't work without these keys anyway, and if we notice these keys are missing, we will have to raise exception and halt everything anyway.

Reply:
Changed